docs/general.md

```diff
@@ -0,0 +1,101 @@
+
+<div class="click-zoom">
+    <label>
+        <input type="checkbox">
+        <img src="../assets/simulation_base.drawio.png" alt="Simulation base section diagram." width="80%" title="Click to zoom in">
+    </label>
+</div>
+
+The detailed UML diagram of quantities and functions defined for `Simulation` is:
+
+<div class="click-zoom">
+    <label>
+        <input type="checkbox">
+        <img src="../assets/simulation.drawio.png" alt="Simulation quantities and functions UML diagram." width="50%" title="Click to zoom in">
+    </label>
+</div>
+
+We use double inheritance from `EntryData` in order to populate the `data` section in the NOMAD archive. You can read more about the public normalize function function and its purpose in our base classes in the corresponding [documentation page](normalize.md). The private function `set_system_branch_depth()` is related with the `ModelSystem` base class, so we refer to its specific [documentation page](model_system/model_system.md) to know more.
+
+## Main sub-sections in `Simulation`
+
+The `Simulation` base class is composed of 4 main sub-sections:
+
+1. `Program`: contains all the program information, e.g., `name` of the program, `version`, etc.
+2. `ModelSystem`: contains all the system information about geometrical positions of atoms, their states, unit cells, symmetry information, etc.
+3. `ModelMethod`: contains all the methodological information, and it is divided in two main aspects: the model Hamiltonian or approximation used in the simulation (e.g., DFT, GW, ForceFields, etc.) and the numerical settings used to compute the properties (e.g., meshes, self-consistent conditions, basis sets, etc.).
+4. `Outputs`: contains all the output properties, as well as references to the `ModelSystem` and `ModelMethod` used to obtain such properties. It might also contain information which will populate `ModelSystem` (e.g., atomic occupations, atomic moments, crystal field energies, etc.).
+
+!!! note "Self-consistent steps, SinglePoint entries, and more complex workflows."
+    In NOMAD, we consider the minimal unit for storing the data in the archive (i.e., an *entry*) as any calculation containing all the self-consistent steps of itself. This is what we call, `SinglePoint`. Thus, we do not split each self-consistent step in its own entry in the NOMAD archive, but rather store them under the same entry in the archive. Any other complex calculation which combines several differentiated self-consistent calculations must be considered a **workflow** (e.g., a `GW` calculation is usually composed of 2 `SinglePoint` entries: the `DFT SinglePoint` self-consistent calculation + the `GW SinglePoint` self-consistent calculations). You can check the corresponding [Github project](https://github.com/nomad-coe/nomad-schema-plugin-simulation-workflow) for more information.
```

Comment:
Similar to above, I would replace "GitHub project" with something like "NOMAD simulation workflow schema"